options.go

```diff
@@ -400,43 +389,98 @@ func WithCGroup(useCGroup bool) Option {
 	})
 }
 
-func WithLoggerLevel(level int) Option {
+func WithLoggerLevel(level Lever) Option {
 	return optionFunc(func(opts *options) (err error) {
 		opts.LogLevel = level
 		return
 	})
 }
 
-type loggerOptions struct {
-	RotateEnable    bool
-	SplitLoggerSize int64 // SplitLoggerSize The size of the log split
+func WithLogger(logger Logger) Option {
+	return optionFunc(func(opts *options) (err error) {
+		opts.loggerLock.Lock()
+		defer opts.loggerLock.Unlock()
+		oldLogger := opts.Logger
+		if oldLogger != nil {
+			switch lg := oldLogger.(type) {
+			case *fileLog:
+				old := lg.logger.Load()
+				if old != nil {
+					oldFd, ok := old.(*os.File)
+					if !ok {
+						//nolint
+						fmt.Println("assert fault")
```

Comment:
```suggestion
						fmt.Println("[Holmes] assert fault, expecting *os.File")
```
we should print more detailed message.